### PR TITLE
Show Course subject by trainee

### PR DESCRIPTION
**Pull request by Rownaknira**

### CODE REVIEW

File: app/views/courses_subjects/edit.html.erb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<div>
+  <h1><%= t ".subj_name" %><%= @courses_subject.subject.name %></h1>
+  <br>
+  <h3><%= t ".subj_desc" %></h3>
+  <p><%= @courses_subject.subject.description %><p>
+  <h3><%= t ".tasks" %></h3>
+  <%= form_for current_user do |f| %>
+    <div class="field form-group">
+        <%= f.label(:title, "Choose from existing Tasks which you have already finished:") %><br>
```
Comment (abdullahallmehedi) [inline app/views/courses_subjects/edit.html.erb]: i18n

File: app/views/supervisor/subjects/_task_fields.erb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<tr class="member">
+  <td>
+    <div class="form-group">
+      <%= f.text_field :name, class: "form-control", placeholder: t('.name_placeholder') %>
```
Comment (abdullahallmehedi) [inline app/views/supervisor/subjects/_task_fields.erb]: "" pls

File: app/views/supervisor/subjects/_task_fields.erb [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    </div>
+  </td>
+  <td>
+    <div class="form-group">
+      <%= link_to_remove_fields t('.remove_button'), f %>
```
Comment (abdullahallmehedi) [inline app/views/supervisor/subjects/_task_fields.erb]: ""

File: app/views/supervisor/subjects/edit.html.erb [hunk 4]
```diff
@@ -0,0 +1,31 @@
+<h1><%= t '.edit' %></h1>
```
Comment (abdullahallmehedi) [inline app/views/supervisor/subjects/edit.html.erb]: ""

File: app/views/supervisor/subjects/edit.html.erb [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+<div class="row">
+  <div class="col-md-6 col-md-offset-3">
+    <%= form_for [:supervisor, @subject] do |f| %>
+      <div class="form-group">
+        <%= f.label :name %>
+        <%= f.text_field :name, class: 'form-control' %>
```
Comment (abdullahallmehedi) [inline app/views/supervisor/subjects/edit.html.erb]: ""

File: app/views/supervisor/subjects/edit.html.erb [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+      </div>
+      <div class="form-group">
+        <%= f.label :description %>
+        <%= f.text_area :description, class: 'form-control'%>
```
Comment (abdullahallmehedi) [inline app/views/supervisor/subjects/edit.html.erb]: ""

File: app/views/supervisor/subjects/edit.html.erb [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+        <%= f.label :tasks %>
+        <table>
+          <tbody>
+            <%= f.fields_for :tasks do |builder| %>
+              <%= render "task_fields", f: builder %>
+            <% end %>
+          </tbody>
+        </table>
+        <div>
+          <%= link_to_add_fields t('.add_more_button'), f, :tasks,
```
Comment (abdullahallmehedi) [inline app/views/supervisor/subjects/edit.html.erb]: ""

File: app/views/supervisor/subjects/edit.html.erb [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+            "btn btn-primary", t('.add_more_button') %>
```
Comment (abdullahallmehedi) [inline app/views/supervisor/subjects/edit.html.erb]: ""

File: app/views/supervisor/subjects/index.html.erb [hunk 9]
```diff
@@ -0,0 +1,25 @@
+<div class="row">
+  <h1 class="text-capitalize"><%= t '.list' %></h1>
```
Comment (abdullahallmehedi) [inline app/views/supervisor/subjects/index.html.erb]: ""

File: app/views/supervisor/subjects/index.html.erb [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+  <br>
+  <table class="table table-striped">
+    <thead>
+      <tr>
+        <th><%= t '.name' %></th>
```
Comment (abdullahallmehedi) [inline app/views/supervisor/subjects/index.html.erb]: ""

File: app/views/supervisor/subjects/index.html.erb [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+        <th><%= t '.desc' %></th>
```
Comment (abdullahallmehedi) [inline app/views/supervisor/subjects/index.html.erb]: ""

File: app/views/supervisor/subjects/index.html.erb [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+        <th colspan="2"></th>
+      </tr>
+    </thead>
+    <%= will_paginate renderer: BootstrapPagination::Rails %>
+    <% @subjects.each do |subject| %>
+      <tr>
+        <td><%= subject.name%></td>
+        <td><%= subject.description%></td>
+        <td><%= link_to t('.edit_link'), edit_supervisor_subject_path(subject) %></td>
```
Comment (abdullahallmehedi) [inline app/views/supervisor/subjects/index.html.erb]: ""

File: app/views/supervisor/subjects/index.html.erb [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+        <td><%= link_to t('.delete_link'), supervisor_subject_path(subject),
```
Comment (abdullahallmehedi) [inline app/views/supervisor/subjects/index.html.erb]: ""

File: app/views/supervisor/subjects/index.html.erb [hunk 14]
```diff
@@ -0,0 +1,25 @@
...
+          method: :delete, data: {confirm: t('.delete_confirm')} %></td>
```
Comment (abdullahallmehedi) [inline app/views/supervisor/subjects/index.html.erb]: ""

File: app/views/supervisor/subjects/index.html.erb [hunk 15]
```diff
@@ -0,0 +1,25 @@
...
+      </tr>
+    <% end %>
+  </table>
+  <br>
+  <%= link_to t('.create'), new_supervisor_subject_path, class: "btn btn-primary" %>
```
Comment (abdullahallmehedi) [inline app/views/supervisor/subjects/index.html.erb]: ""

File: app/views/supervisor/subjects/new.html.erb [hunk 16]
```diff
@@ -0,0 +1,31 @@
+<h1><%= t '.add' %></h1>
```
Comment (abdullahallmehedi) [inline app/views/supervisor/subjects/new.html.erb]: ""

File: app/views/supervisor/subjects/new.html.erb [hunk 17]
```diff
@@ -0,0 +1,31 @@
...
+<div class="row">
+  <div class="col-md-6 col-md-offset-3">
+    <%= form_for [:supervisor, @subject] do |f| %>
+      <div class="form-group">
+        <%= f.label :name %>
+        <%= f.text_field :name, class: 'form-control' %>
```
Comment (abdullahallmehedi) [inline app/views/supervisor/subjects/new.html.erb]: ""

File: app/views/supervisor/subjects/new.html.erb [hunk 18]
```diff
@@ -0,0 +1,31 @@
...
+      </div>
+      <div class="form-group">
+        <%= f.label :description %>
+        <%= f.text_area :description, class: 'form-control'%>
```
Comment (abdullahallmehedi) [inline app/views/supervisor/subjects/new.html.erb]: ""

File: app/views/supervisor/subjects/new.html.erb [hunk 19]
```diff
@@ -0,0 +1,31 @@
...
+        <%= f.label :tasks %>
+        <table>
+          <tbody>
+            <%= f.fields_for :tasks do |builder| %>
+              <%= render "task_fields", f: builder %>
+            <% end %>
+          </tbody>
+        </table>
+        <div>
+          <%= link_to_add_fields t('.add_more_in_add'), f, :tasks,
```
Comment (abdullahallmehedi) [inline app/views/supervisor/subjects/new.html.erb]: ""

File: app/views/supervisor/subjects/new.html.erb [hunk 20]
```diff
@@ -0,0 +1,31 @@
...
+            "btn btn-primary", t('.add_more_in_add') %>
```
Comment (abdullahallmehedi) [inline app/views/supervisor/subjects/new.html.erb]: ""

File: config/locales/en.yml [hunk 21]
```diff
@@ -1,15 +1,19 @@
...
-  tasks: "Tasks:"
   button:
     hide: Hide
     show: Show
+  confirm_message: You sure?
   contact: Contact
+  courses_subjects:
+    edit:
+      subj_desc: Description
+      subj_name: 'Subject Name: '
```
Comment (abdullahallmehedi) [inline config/locales/en.yml]: ""

File: config/locales/en.yml [hunk 22]
```diff
@@ -1,15 +1,19 @@
...
+      tasks: 'Tasks: '
```
Comment (abdullahallmehedi) [inline config/locales/en.yml]: ""

File: config/locales/en.yml [hunk 23]
```diff
@@ -70,7 +74,23 @@ en:
...
+        edit_link: Edit
+        list: Listing of Subjects
+      new:
+        add: Add Subject
+        add_more_in_add: Add new task
+      task_fields:
+        name_placeholder: Task Name
+        remove_button: Remove a task
   system_name: Framgia Training Management System
+  tasks: 'Tasks:'
```
Comment (abdullahallmehedi) [inline config/locales/en.yml]: ""

**Comment by rofiuddin** [comment]
okay

File: app/controllers/supervisor/subjects_controller.rb [hunk 24]
```diff
@@ -3,6 +3,7 @@ class Supervisor::SubjectsController < ApplicationController
   load_and_authorize_resource :course
 
   def index
+    @subjects = Subject.all.paginate page: params[:page], per_page: 15
```
Comment (dieunb) [inline app/controllers/supervisor/subjects_controller.rb]: remove `.all`

File: app/controllers/users_controller.rb [hunk 25]
```diff
@@ -9,8 +9,20 @@ def show
     @activities = @user.activities.paginate page: params[:page]
   end
 
+  def update
+    if @user.update_attributes user_params
+      redirect_to course_path
+    else
+      render "courses_subjects/edit"
```
Comment (dieunb) [inline app/controllers/users_controller.rb]: i don't know why???

File: app/views/courses_subjects/edit.html.erb [hunk 26]
```diff
@@ -0,0 +1,17 @@
+<div>
+  <h1><%= t ".subj_name" %><%= @courses_subject.subject.name %></h1>
+  <br>
+  <h3><%= t ".subj_desc" %></h3>
+  <p><%= @courses_subject.subject.description %><p>
+  <h3><%= t ".tasks" %></h3>
+  <%= form_for current_user do |f| %>
+    <div class="field form-group">
+        <%= f.label(:title, t ".choose" ) %><br>
```
Comment (dieunb) [inline app/views/courses_subjects/edit.html.erb]: indent

File: app/views/courses_subjects/edit.html.erb [hunk 27]
```diff
@@ -0,0 +1,17 @@
...
+        <%= f.collection_check_boxes(:task_ids, @courses_subject.tasks, :id, :name) do |b| %>
+          <%= b.label(class: "checkbox-inline") { b.check_box + b.text } %>
```
Comment (dieunb) [inline app/views/courses_subjects/edit.html.erb]: `{b.check_box + b.text}`

File: app/controllers/users_controller.rb [hunk 28]
```diff
@@ -9,8 +9,20 @@ def show
     @activities = @user.activities.paginate page: params[:page]
   end
 
+  def update
```
Comment (dieunb) [inline app/controllers/users_controller.rb]: Task auto assign to user when super assign they to course

Comment (Rownaknira) [inline app/controllers/users_controller.rb]: when trainee finishes any task, it will be used to update tasklist of that trainee

**Pull request merged**